cmd/snap-exec/main.go

```diff
@@ -141,16 +141,34 @@ func snapExecApp(snapApp, revision, command string, args []string) error {
 	if err != nil {
 		return err
 	}
-	// strings.Split() is ok here because we validate all app fields
-	// and the whitelist is pretty strict (see
-	// snap/validate.go:appContentWhitelist)
-	cmdArgv := strings.Split(cmdAndArgs, " ")
-	cmd := cmdArgv[0]
-	cmdArgs := cmdArgv[1:]
 
 	// build the environment from the yaml
 	env := append(os.Environ(), osutil.SubstituteEnv(app.Env())...)
 
+	// strings.Split() is ok here because we validate all app fields
+	// and the whitelist is pretty strict (see
+	// snap/validate.go:appContentWhitelist)
+	tmpCmdArgv := strings.Split(cmdAndArgs, " ")
+	cmd := tmpCmdArgv[0]
+
+	cmdArgs := make([]string, 0, len(tmpCmdArgv[1:]))
```

Comment:
I'd feel better if this was a separate function that can be tested in isolation.